### ISSUE TITLE
Time series Normalization Factor = 0

### ISSUE THREAD

**Issue by SebastianManriqueM** [NREL-Sienna/InfrastructureSystems.jl#414]
When using "normalization_factor": "Max", it takes the maximum value of the time series as a normalization factor. 

However, if all the values of a time series are 0, then it results in a time series loaded as "NaN". In such cases, this is supposed to be detected and handled to avoid time series filled wit  "NaN".

Could be possible to add an option to normalize taking the _active_power_limits__?

**Comment by daniel-thom** [comment]
What is the desired outcome if the user chooses `Max` and all values are 0? Should the code throw an error?

**Comment by daniel-thom** [comment]
@pesap

**Comment by pesap** [comment]
Yes. I think it should throw an error. There is no sense of adding a time series with all zeros.

**Comment by daniel-thom** [comment]
@SebastianManriqueM Your secondary request to use active_power_limits as the normalization factor needs more discussion. We apply a normalization factor in two cases: (1) construction of SingleTimeSeries/Deterministic and (2) table data parser.

In the first case I think that you could pass the normalization factor when you create the time series and attach it to a component.

In the second case, if that is important, the time_series_pointers CSV/JSON file would have to include some metadata about how to extract an accessor function to use for the normalization factor.

What are you thinking?

**Comment by daniel-thom** [comment]
@SebastianManriqueM @jd-lara I'm still confused on what to do with time series of all 0. If the user asks for `normalization_factor = max` and all values are 0, the situation is untenable. We have to do one of these:
1. Throw an exception (added in this PR).
2. Julia throws "division by zero" error (behavior in the current version).
3. Ignore normalization factor and store zeros.

Which one should I go with?

Secondly, in both the current version and this MR, it is OK for the user to store zeros in the time series if they do not set normalization_factor. Is this OK?

On the question of allowing active_power_limits as the the normalization factor in the "time series pointers" file: One entry looks like this:
```
    {
        "simulation": "DAY_AHEAD",
        "resolution": 3600,
        "category": "Generator",
        "component_name": "122_HYDRO_1",
        "module": "PowerSystems",
        "type": "SingleTimeSeries",
        "name": "max_active_power",
        "scaling_factor_multiplier": "get_max_active_power",
        "scaling_factor_multiplier_module": "PowerSystems",
        "normalization_factor": 52.49761899362675,
        "data_file": "./RTS_GMLC_forecasts/gen/Hydro/DAY_AHEAD_hydro.csv"
    },
```
We would have to add something like this:
```
        "normalization_factor": "get_active_power_limits"
```
But then that doesn't get us all the way there. That method returns a tuple of min and max. How do we generically represent this action that will work for any type and method? Also, Jose has indicated that this style of loading time series will soon be deprecated.
